jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/events/ConfigurationChangeEvent.java

```diff
@@ -0,0 +1,37 @@
+package com.googlecode.jmxtrans.cluster.events;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * ConfigurationChangeEvent. This class is passed to all the registered ClusterStateChangeListener. It contains
+ * the main events relating to the cluster connection.
+ *
+ * @author Tibor Kulcsar
+ * @since <pre>May 17, 2016</pre>
+ */
+@ToString(includeFieldNames=true, exclude = "configuration")
+public class ConfigurationChangeEvent {
+    @Getter
+    private final ConfigurationChangeEvent.Type type;
+
+    @Getter
+    private final String[] configuration;
+
+    public ConfigurationChangeEvent(ConfigurationChangeEvent.Type type, String[] configuration) {
+        this.type = checkNotNull(type, "The change type cannot be null!");
```

Comment:
I prefer tagging params with `@NonNull` and let static analysis check them at compile time rather than adding preconditions. Preconditions make sense when exposing methods to external clients (clients that we don't control), but I don't think this is the case here.